x/group/types.go

```diff
@@ -74,14 +74,23 @@ func (p ThresholdDecisionPolicy) Allow(tallyResult TallyResult, totalPower strin
 	if err != nil {
 		return DecisionPolicyResult{}, err
 	}
-	if yesCount.Cmp(threshold) >= 0 {
-		return DecisionPolicyResult{Allow: true, Final: true}, nil
-	}
 
 	totalPowerDec, err := math.NewNonNegativeDecFromString(totalPower)
 	if err != nil {
 		return DecisionPolicyResult{}, err
 	}
+
+	// the real threshold of the policy is `min(threshold,total_weight)`. If
+	// the group member weights changes (member leaving, member weight update)
+	// and the threshold doesn't, we can end up with threshold > total_weight.
+	// In this case, as long as everyone votes yes (in which case
+	// `yesCount`==`realThreshold`), then the proposal still passes.
+	realThreshold := min(threshold, totalPowerDec)
```

Comment:
@amaurym here can we just check if everyone voted yes instead of checking against realThreshold?

Reply:
I see. There are 2 scenarios:
- if threshold>totalPower, then we should check that everyone votes yes
- if threshold<totalPower, we don't need everyone to vote yes

In the 2nd cases, we don't check everyone voted yes. The `realThreshold` is used to share logic to avoid having an `if then else` case, but I can introduce that if it's clearer.